packages/medusa-plugin-auth/src/core/validate-callback.ts

```diff
@@ -92,6 +104,18 @@ export async function validateStoreCallback<
 				});
 			}
 
+			if (
+				hasEmailVerifiedField &&
+				customer.metadata &&
+				customer.metadata[CUSTOMER_METADATA_KEY] &&
+				!customer.metadata[EMAIL_VERIFIED_KEY]
```

Comment:
In case the value is false it will always update the customer, should we also check that the property value is different than the verified value? Wdyt?

Reply:
@adrien2p good point that probably makes sense to avoid unnecessary updates.
However, I realized yesterday (from using this on my own application) that the emailVerified is not updated until the user is logged out and logged back in again. Because it has to be re-fetched from Auth0, as far as I know.

So I was considering closing this PR, because I don't think it makes sense unless the value is always up-to-date. Maybe you have a different idea?

Reply:
Usually once the email is verified, it does not change anymore. We can keep it as it is I suppose and you get an extra update in case it is undefined or false. Is that ok?